core/src/main/java/org/elasticsearch/action/support/TransportAction.java

```diff
@@ -66,7 +66,7 @@ protected TransportAction(Settings settings, String actionName, ThreadPool threa
         return future;
     }
 
-    public final void execute(Request request, ActionListener<Response> listener) {
+    public final Task execute(Request request, ActionListener<Response> listener) {
         Task task = taskManager.register("transport", actionName, request);
         if (task == null) {
```

Comment:
@bleskes I added an option for an action to [opt out](https://github.com/elastic/elasticsearch/blob/d9a01225cb668c1764d10c9ddd5f4318c5057218/core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java#L588) from task management tracking. If the [TransportRequest#createTask](https://github.com/elastic/elasticsearch/blob/a89dba27c25d02f7b7c01bf569a29588392f6f48/core/src/main/java/org/elasticsearch/transport/TransportRequest.java#L48) method returns null instead of a task object, this action is not tracked by the task manager and the `register` call returns null. 

I forgot to document this behavior. Thanks for the reminder. I will add comments in a bit.